packages/flutter_tools/lib/src/build_system/targets/dart_plugin_registrant.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:package_config/package_config.dart';
+
+import '../../dart/package_map.dart';
+import '../../base/file_system.dart';
+import '../../plugins.dart';
+import '../../project.dart';
+import '../build_system.dart';
+import 'common.dart';
+
+/// Based on the current dependency map in `pubspec.lock`. Generated a new `generated_main.dart`
+/// and replace the `./dart_tool/flutter_build/generated_main.dart`.
+class DartPluginRegistrantTarget extends Target {
+
+  /// Construct a [DartPluginRegistrantTarget].
+  ///
+  /// If `project` is unset, a [FlutterProject] based on environment is used.
+  DartPluginRegistrantTarget({FlutterProject project}) {
```

Comment:
the project param is mainly configurable for testing. In tests, i needed a `FlutterProject.fromDirectoryTest`

Reply:
You can add a @visibleForTesting annotation on it if it isn't meant to be normally used